expr/scratch.go

```diff
@@ -0,0 +1,60 @@
+package expr
+
+import (
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zng"
+)
+
+type scratch struct {
+	keep bool
+	buf  zcode.Bytes
+}
+
+func (s *scratch) Int(v int64) zcode.Bytes {
+	if s.keep {
+		return zng.EncodeInt(v)
+	}
+	b := s.buf
+	if b != nil {
+		b = b[:0]
+	}
+	s.buf = zng.AppendInt(b, v)
```

Comment:
You do not need all these nil checks in all these functions:
```suggestion
	s.buf = s.buf[:0]
	s.buf = zng.AppendInt(s.buf, v)
```